zrml/lsd-lmsr/src/lib.rs

```diff
@@ -0,0 +1,76 @@
+//! # Court
+//!
+//! Manages market disputes and resolutions.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+extern crate alloc;
+
+pub mod constants;
+mod mock;
+mod tests;
+pub mod traits;
+pub mod types;
+pub use pallet::*;
+
+#[frame_support::pallet]
+mod pallet {
+    use core::marker::PhantomData;
+    use frame_support::{
+        pallet_prelude::{MaybeSerializeDeserialize, Member},
+        traits::{Hooks, Time},
+        Parameter,
+    };
+    use parity_scale_codec::Codec;
+    use sp_runtime::traits::AtLeast32BitUnsigned;
+    use sp_std::fmt::Debug;
+    use substrate_fixed::types::extra::LeEqU128;
+
+    #[pallet::config]
+    pub trait Config: frame_system::Config {
+        /// Balance type: Defines the type of traded amounts
+        type Balance: Parameter
+            + Member
+            + AtLeast32BitUnsigned
+            + Codec
+            + Default
+            + Copy
+            + MaybeSerializeDeserialize
+            + Debug;
+
+        /// The type that offers timestamping functionality
+        type Timestamp: Time;
+
+        /// The type that will be used for the fractional part of the fixed point numbers
+        type FractionalType: LeEqU128;
```

Comment:
I was trying to the deduce the type that determines the number of fractional bits from the number of one unit of token (in our case BASE) automatically and use it in storage, but failed after thinking through multiple options. I think since this number should not change during the whole lifetime of the chain it is okay to supply it this way.

Reply:
This type would be beneficial if two different runtimes are using two different `FractionalType`. For example, for some reason polkadot's runtime is using `FractionalType = U30` and kusama's runtime is using `FractionalType = U24` or we only have one runtime but want to test different fractions inside a pallet's mock (can't imagine why this would be necessary).

In my opinion, I think it is probably better to stick with a fixed fraction